legacy_publisher/dita_publisher.py

```diff
@@ -90,45 +92,109 @@ def publish_regions(regions=None, sourcepath=None):
+    for richcollection in nstcountries:
+        rich_region = richcollection.title
+        process_countries(region=rich_region, current=rich_region, nst=True, richcollection=richcollection)
 
-        body = create_body(root=root,topic=topic)
 
-        number_country = 0
-        for country in countries:
-            if(current_region == country.region.region):
-                number_country+=1
 
-        row = create_table(root=root,body=body, cols=str(number_country))
+def get_countries(regions=None, stcountries=None):
+    countries = []
+    for region in regions.regions:
+        current_region = region.region
+        for country in stcountries:            
+            if current_region not in countries and current_region == country.region.region:
+                countries.append(current_region)
+
+    return countries
 
-        export_dita = dirname(DITA_REGIONS_EXPORT_FILE)+"/"+current_region+"/"+current_region+".dita"
+def process_countries(region=None, countries=None,current=None,nst=False, richcollection=None):
+    current_region=current
 
+    number_country = 0
+    if countries != None:
         for country in countries:
             if(current_region == country.region.region):
-                entry = root.createElement('entry')
-                relative_path_utl = "#" if country.url == None else os.path.relpath(country.url , dirname(export_dita))
-                xref = create_xref(root=root,text=country.country,url=relative_path_utl,format="html")
-                entry.appendChild(xref) 
-                row.appendChild(entry)
+                number_country+=1
 
-        isDestExist = os.path.exists(dirname(export_dita))
-        if not isDestExist:
-            os.makedirs(dirname(export_dita))
-        print("export_dita :", export_dita)
-        write_dita_doc(root, export_dita)
+    pstr = "regions/"
+    export_dita = dirname(DITA_REGIONS_EXPORT_FILE)+"/"+pstr+current_region+"/"+current_region+".dita"
+    root = create_nst_page(current_region=current_region,export_dita=export_dita, richcollection=richcollection) if nst == True else create_page(current_region=current_region,number_country=number_country,countries=countries,export_dita=export_dita)
 
-        if(dita_ot != None):
-            xml_file = abspath(export_dita)
-            dtd_file = dita_ot+'/plugins/org.oasis-open.dita.v1_2/dtd/technicalContent/dtd/topic.dtd'
-            validate(xml_file, dtd_file)
+    doc_str = doctype_richcollection_str if nst == True else None
 
+    isDestExist = os.path.exists(dirname(export_dita))
+    if not isDestExist:
+        os.makedirs(dirname(export_dita))
+    print("export_dita :", export_dita)
+    write_dita_doc(root, export_dita, doctype_str=doc_str)
 
+    if(dita_ot != None):
+        xml_file = abspath(export_dita)
+        dtd_file = dita_ot+'/plugins/org.oasis-open.dita.v1_2/dtd/technicalContent/dtd/topic.dtd'
+        validate(xml_file, dtd_file)
```

Comment:
in the above we are validating against `topic.dtd`. Shouldn't it be `rich-collection.dtd`?